WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListFragment.java

```diff
@@ -769,4 +770,22 @@ public void onEventMainThread(UploadService.UploadMediaSuccessEvent event) {
         }
     }
 
+    @SuppressWarnings("unused")
+    public void onEventMainThread(UploadService.UploadMediaRetryEvent event) {
+        if (!isAdded()) {
+            return;
+        }
+
+        if (event.mediaModelList != null && !event.mediaModelList.isEmpty()) {
+            // if there' a Post to which the retried media belongs, clear their status
+            Set<PostModel> postsToRefresh = PostUtils.getPostsThatIncludeThisMedia(mPostStore, event.mediaModelList);
+            // now that we know which Posts  to refresh, let's do it
+            for (PostModel post : postsToRefresh) {
+                if (post != null) {
```

Comment:
I don't think `post` can be null here, but it is possible that `getPositionForPost()` will return -1. So this could be rewritten as:

```
for (PostModel post : postsToRefresh) {
    int position = getPostListAdapter().getPositionForPost(post);
    if (position > -1) {
        getPostListAdapter().notifyItemChanged(position);
    }
}
```

Reply:
nice! changed here bc089e1